gensim/models/experimental/custom_callbacks.py

```diff
@@ -0,0 +1,42 @@
+import logging
+try:
+    from keras.callbacks import Callback
+    KERAS_AVAILABLE = True
+except ImportError:
+    KERAS_AVAILABLE = False
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    format='%(asctime)s : %(levelname)s : %(message)s', level=logging.INFO)
+
+
+class ValidationCallback(Callback):
+
+    def __init__(self, test_data):
+        if not KERAS_AVAILABLE:
+            raise ImportError("Please install Keras to use this class")
+
+        self.test_data = test_data
+
+    def on_epoch_end(self, epoch, logs={}):
+        # Import has to be here to prevent cyclic import
+        from evaluation_metrics import mapk, mean_ndcg
+        X1 = self.test_data["X1"]
```

Comment:
what's the type of `test_data`?  Is this dict?

Reply:
done